cmd/gcp-controller-manager/main.go

```diff
@@ -224,7 +224,7 @@ func run(s *controllerManager) error {
 		panic("unreachable")
 	}
 
-	run(nil)
+	run(ctx)
```

Comment:
Since you're cleaning stuff up I think this variable name is gross. Basically there is:
```go
func run(s *controllerManager) {
  ...
  run := func(ctx context.Context) {...}
  ...
  run(ctx)
}
```
I was about to write out a comment how the param is the incorrect type and the recursion is bad. But its just confusing variable names. Could you rename the func to something different than the surrounding func's name?

Reply:
Done